source/slang/slang-ir-use-uninitialized-out-param.cpp

```diff
@@ -75,27 +75,33 @@ namespace Slang
                 }
             }
             // Check all address loads.
-            List<IRLoad*> loads;
+            List<IRInst*> loadsAndReturns;
```

Comment:
Now that I think of it, we should expand this pass to cover `Var`s too. Since this pass take place after SSA pass, normal scalar vars are already lowered into SSA values. The only vars remaining are arrays and struct typed variables. The check for them is the same for `out` params.

In a separate pass we should also check all uses of `kIROp_Undef` and warn/error on those. Perhaps there are a couple valid use of `undefined` values and we need to skip those.

Reply:
https://github.com/shader-slang/slang/issues/2817